apm-agent-api/src/main/java/co/elastic/apm/api/SpanImpl.java

```diff
@@ -56,15 +56,22 @@ public Span addTag(String key, String value) {
 
     @Nonnull
     @Override
-    public Span addTag(String key, Number value) {
-        doAddNumberTag(key, value);
+    public Span addLabel(String key, String value) {
+        doAddTag(key, value);
```

Comment:
Why not change `doAddTag` to `doAddLabel`? It means we will have duplicated instrumentation (for both), but better lose any unnecessary `tag` reminders.

Reply:
I'd like to avoid having different instrumentation for different API versions if possible. The problem is that we can't easily unit test for previous API versions. We could potentially have a separate test module for each API version we want to test but I don't think it's worth the complexity.

Reply:
What's the problem of testing? The old API is still there, only deprecated. You can write tests using it.

Reply:
you're right. done.